### ISSUE TITLE
About priority_search and backjumps

### ISSUE THREAD

**Issue by raphaelboudreault** [chuffed/chuffed#76]
Hi! I'm currently working with MiniZinc and Chuffed to solve an industrial scheduling problem. The use of `priority_search` has made a huge improvement on the solving time, specifying a specific intelligent search heuristic for our problem.

That said, I was experimenting some search heuristics with `priority_search`, and encountered a strange behavior. Given an array X[1..n] of int variables, these two heuristics lead to completely different results: 
```
int_search(X, smallest, indomain_min, complete);
priority_search(X, [int_search([X[i]], input_order, indomain_min, complete) | i in 1..n], smallest, complete);
```
I was suprised to observe a difference, since the idea is the same in the two cases, select the variable with the smallest value in its domain, and branch on the smallest possible value for that variable. Logicly, I would have expected the same behavior.

Looking closer to the search tree and Chuffed source code, the difference happens when a conflict/backjump occurs. When it happens, the choice of the branching group is not reevaluated given the variable selection (_smallest_ in this case). Thus, it is possible to make the decision `X[i] = v` where, in fact, there exists another variable in `X` with a possible value smaller than `v`. This case doesn't happen with the `int_search` heuristic, since there is only one branching group.

In short, is this the correct behavior for the `priority_search`? If not, the "fix" I propose would be to make `PriorityBranchGroup` a `BranchGroup` with `terminal = true`, since we want the search to re-evaluate the branch group selection everytime.
